lib/node_modules/@stdlib/plot/unicode/static_plots/base/lib/plotbase.js

```diff
@@ -0,0 +1,456 @@
+var getpadLeft = require( './props/padLeft/get.js' );
+var setpadRight = require( './props/padRight/set.js' );
+var getpadRight = require( './props/padRight/get.js' );
+var Pallete = require( './pallete.js' );
+var defaults = require( './default.js' );
+
+
+// VARIABLES //
+
+// List of private properties (note: keep in alphabetical order):
+var PRIVATE_PROPS = [
+	'_cols',
+	'_padBottom',
+	'_padLeft',
+	'_padRight',
+	'_padTop',
+	'_rows',
+	'_setSize'
+];
+
+
+// FUNCTIONS //
+
+var merge = mergeFcn({
+	'extend': false
+});
+
+
+// MAIN //
+
+/**
+* PlotBase constructor.
+*
+* @constructor
+* @param {Options} options - constructor options
+* @param {number} [Options.padBottom] - padding bottom
+* @param {number} [Options.padLeft] - padding left
+* @param {number} [Options.padRight] - padding right
+* @param {number} [Options.padTop] - padding top
+* @param {Object} [Options.setSize] - size object with character width and height
+* @throws {TypeError} must provide valid options
+* @returns {PlotBase} PlotBase instance
+*
+* @example
+* var PlotBase = new PlotBase();
+*/
+function PlotBase( ) {
+	var options;
+	var nargs;
+	var keys;
+	var opts;
+	var key;
+	var i;
+
+	defineProperty( this, '_x', {
+		'configurable': false,
+		'enumerable': false,
+		'writable': false,
+		'value': []
+	});
+
+	defineProperty( this, '_y', {
+		'configurable': false,
+		'enumerable': false,
+		'writable': false,
+		'value': []
+	});
+
+	defineProperty( this, '_marker', {
+		'configurable': false,
+		'enumerable': false,
+		'writable': false,
+		'value': []
+	});
+
+	defineProperty( this, '_color', {
+		'configurable': false,
+		'enumerable': false,
+		'writable': false,
+		'value': []
+	});
+
+	defineProperty( this, '_palt', {
+		'configurable': false,
+		'enumerable': false,
+		'writable': false,
+		'value': new Pallete(this)
+	});
+
+	for ( i = 0; i < PRIVATE_PROPS; i++ ) {
+		defineProperty( this, PRIVATE_PROPS[ i ], {
+			'configurable': false,
+			'enumerable': false,
+			'writable': true,
+			'value': null
+		});
+	}
+
+	nargs = arguments.length;
+	opts = defaults();
+	if ( nargs ) {
+		options = arguments[ 0 ];
+	}
+	else {
+		options = {};
+	}
+	opts = merge( opts, options );
+	keys = objectKeys( opts );
+	for ( i = 0; i < keys.length; i++ ) {
+		key = keys[ i ];
+		this[ key ] = opts[ key ];
+	}
+	return this;
+}
+
+/**
+ * Add all the ansi codes to the matrix.
+ *
+ * @name build
+ * @memberof PlotBase.prototype
+ * @type {Function}
+ * @returns {void}
+ *
+ */
+setNonEnumerableReadOnly(PlotBase.prototype, 'build', function build() {
+	var finalC =[];
+	var temp;
+	var i;
+	var j;
+
+	for ( i = 0; i < this._x.length; i++ ) {
+		finalC.push( fillCord( this._x[ i ], this._y[ i ] ) );
+	}
+
+	for ( i = 0; i < finalC.length; i++ ) {
+		temp = finalC[ i ];
+		for ( j = 0; j < temp.length; j++ ) {
+			this._palt.setElement( temp[ j ][ 0 ], temp[ j ][ 1 ], this._marker[ i ], this._color[ i ] );
+		}
+	}
+});
+
+/**
+ * Display the rendered plots.
```

Comment:
We have many methods for "rendering". Why `build`, `show`, `draw`, `drawFrame`, `draw*`, and `render` (elsewhere)? It is not clear how these all fit together and why it makes sense to organize the code like this.